src/unix/fs.c

```diff
@@ -291,25 +299,7 @@ static ssize_t uv__fs_read(uv_fs_t* req) {
     if (no_preadv) retry:
 # endif
     {
-      off_t nread;
-      size_t index;
-
-      nread = 0;
-      index = 0;
-      result = 1;
-      do {
-        if (req->bufs[index].len > 0) {
-          result = pread(req->file,
-                         req->bufs[index].base,
-                         req->bufs[index].len,
-                         req->off + nread);
-          if (result > 0)
-            nread += result;
-        }
-        index++;
-      } while (index < req->nbufs && result > 0);
-      if (nread > 0)
-        result = nread;
+      result = pread(req->file, req->bufs[0].base, req->bufs[0].len, req->off);
```

Comment:
Shouldn't we go through each of the buffers here?

Reply:
If the caller wants all of the data, they are required to implement that loop, so I opted here for the simpler solution of having this function do exactly one syscall on all code-paths, and returning that one result. I reference this behavior in the PR commit message:

> For simplicity and predictability (since the user must handle the retry
anyways), always emit exactly one readv/pread/preadv syscall and return
that result to the user.

The alternative would be that we be that we loop here until we encounter an error, then discard the error and return the result up to that point. That's valid behavior (it's what the kernel internally is expected to do), but it also seemed to me like more complexity than necessary. It's also not clear to me whether it is more correct to fill each buffer only as much as one `read` call returns (but to fill as many buffers as possible—as it does not), or to fill each buffer fully (as it would do in all other cases in this code). Handling that here seems like it would add yet further complexity.

Reply:
It makes sense to me. Thanks!